libs/feature/search/src/lib/state/effects.ts

```diff
@@ -72,20 +78,31 @@ export class SearchEffects {
     )
   )
 
+  private actionsWithNewResults$ = this.actions$.pipe(
+    ofType(
+      SET_SORT_BY,
+      SET_FILTERS,
+      UPDATE_FILTERS,
+      SET_SEARCH,
+      SET_FAVORITES_ONLY,
+      SET_SPATIAL_FILTER_ENABLED,
+      PAGINATE,
+      SET_PAGE_SIZE
+    )
+  )
+
   requestNewResults$ = createEffect(() =>
-    this.actions$.pipe(
-      ofType(
-        SET_SORT_BY,
-        SET_FILTERS,
-        UPDATE_FILTERS,
-        SET_SEARCH,
-        SET_FAVORITES_ONLY,
-        SET_SPATIAL_FILTER_ENABLED,
-        PAGINATE,
-        SET_PAGE_SIZE
-      ),
-      debounceTime(0),
-      map((action: SearchActions) => new RequestNewResults(action.id))
+    this.actionsWithNewResults$.pipe(
+      // this will aggregate actions until the debounceTime ticks
+      buffer(this.actionsWithNewResults$.pipe(debounceTime(0))),
```

Comment:
Why do we need this `buffer/debounceTime`?

Reply:
the idea here is that if you call for instance `searchFacade.orderBy('createDate').setPageSize(10)` etc., each call to the facade will trigger an action that will eventually trigger a `RequestNewResults` action. So you end up having many of these, many search requests and potential conflicts.

But because there are several search ids, the grouping is actually quite tricky indeed... I've tried to make this as clear as possible :grimacing: